### DISCUSSION TITLE
beforeScreenshot hook in previewjs/screenshot

### DISCUSSION THREAD

**Discussion by KraKeN-47** [fwouts/previewjs#1559] [Q&A] [answered]
Hello,
is it possible to apply some logic before making a screenshot similair to `beforeScreenshot` like in viteshot? Or is this all up to the user?

**Comment by fwouts** [comment]
Hey @KraKeN-47, if Preview.js were to support [Storybook's play functions](https://storybook.js.org/docs/react/writing-stories/play-function) and run them before the screenshot is taken, would that work for you? That doesn't work yet, but looks like it might be easier to implement than I expected.

**Reply by fwouts** [reply, marked as answer]
This is now implemented in v0.1.2.

All you need is to define a function `play` on your story (or even on a component) which looks like this:
```js
MyStory.play = ({ canvasElement }) => {
  // Do something with canvasElement, which is the root of the DOM.
  // See https://storybook.js.org/docs/react/writing-stories/play-function for a Storybook example
  // or directly use @testing-library/dom like https://testing-library.com/docs/dom-testing-library/example-intro
}
```
